app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,328 @@
+      //   }
+      //   clientGeneratedNotifications.push(notification);
+      // };
+
+      var iconClassByEventSeverity = {
+        Normal: 'pficon pficon-info',
+        Warning: 'pficon pficon-warning-triangle-o'
+      };
+
+      angular.extend(drawer, {
+        drawerHidden: true,
+        allowExpand: true,
+        drawerExpanded: false,
+        drawerTitle: 'Notifications',
+        hasUnread: false,
+        showClearAll: true,
+        showMarkAllRead: true,
+        onClose: function() {
+          drawer.drawerHidden = true;
+        },
+        onMarkAllRead: function(group) {
+          _.each(group.notifications, function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+          });
+          render();
+          $rootScope.$emit('NotificationDrawerWrapper.onMarkAllRead');
+        },
+        onClearAll: function(group) {
+          _.each(group.notifications, function(notification) {
+            EventsService.markRead(notification.event);
+            EventsService.markCleared(notification.event);
+          });
+          group.notifications = [];
+          render();
+          $rootScope.$emit('NotificationDrawerWrapper.onMarkAllRead');
+        },
+        notificationGroups: notificationGroups,
+        headingInclude: 'views/directives/notifications/header.html',
+        notificationBodyInclude: 'views/directives/notifications/notification-body.html',
+        customScope: {
+          clear: function(notification, index, group) {
+            EventsService.markCleared(notification.event);
+            group.notifications.splice(index, 1);
+            countUnreadNotificationsForAllGroups();
+          },
+          markRead: function(notification) {
+            notification.unread = false;
+            EventsService.markRead(notification.event);
+            countUnreadNotificationsForAllGroups();
+          },
+          getNotficationStatusIconClass: function(event) {
+            return iconClassByEventSeverity[event.type] || iconClassByEventSeverity.info;
+          },
+          getStatusForCount:  function(countKey) {
+            return iconClassByEventSeverity[countKey] || iconClassByEventSeverity.info;
+          },
+          close: function() {
+            drawer.drawerHidden = true;
+          }
+        }
+      });
+
+      var projectChanged = function(next, current) {
+        return _.get(next, 'params.project') !== _.get(current, 'params.project');
+      };
+
+      var reset = function() {
+        getProject($routeParams.project).then(function() {
+          watchEvents($routeParams.project, eventWatchCallback);
+          //watchNotifications($routeParams.project, notificationWatchCallback);
+          hideIfNoProject($routeParams.project);
+          render();
+        });
+      };
+
+      drawer.$onInit = function() {
+        if(DISABLE_GLOBAL_EVENT_WATCH || INSUFFICIENT_WATCHES || !AuthService.isLoggedIn()) {
```

Comment:
@spadgett how do you feel about this?  Added `!AuthService.isLoggedIn()` here to guard against future pages that may be public & just stop the directive from doing anything.

Same check in the counter.